mlflow/transformers/flavor_config.py

```diff
@@ -157,6 +165,65 @@ def _get_instance_type(obj):
     return obj.__class__.__name__
 
 
+def build_flavor_config_from_local_checkpoint(
+    local_checkpoint_dir: str,
+    task: str,
+    processor=None,
+    torch_dtype=None,
+) -> Dict[str, Any]:
+    """
+    Generates the flavor metadata from a Hugging Face model repository ID
+    e.g. "meta-llama/Meta-Llama-3.1-405B, instead of the pipeline instance in-memory.
+    """
+    from transformers import AutoTokenizer, pipelines
+    from transformers.utils import is_torch_available
+
+    from mlflow.transformers.model_io import _MODEL_BINARY_FILE_NAME
+
+    config_path = os.path.join(local_checkpoint_dir, "config.json")
+    if not os.path.exists(config_path):
+        raise MlflowException(
+            f"The provided directory {local_checkpoint_dir} does not contain the config.json file."
+            "Please ensure that the directory contains a valid transformers model checkpoint.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+    with open(config_path) as f:
+        config = json.load(f)
+
+    task_metadata = pipelines.check_task(task)
```

Comment:
Is this always a 1:1 mapping?

Reply:
Yes! https://github.com/huggingface/transformers/blob/9230d78e76611cfa38c845213021aeb185362d10/src/transformers/pipelines/__init__.py#L156-L427